src/opening-period/form/OpeningPeriodForm.tsx

```diff
@@ -0,0 +1,322 @@
+import React, { useState } from 'react';
+import { ArrayField, useFieldArray, useForm } from 'react-hook-form';
+import { useHistory } from 'react-router-dom';
+import {
+  DatePeriod,
+  DatePeriodOptions,
+  FormWeekdays,
+  Language,
+  ResourceState,
+  ResourceStateApiOption,
+  ResourceStateOption,
+  TimeSpan as TimeSpanApiFormat,
+  TimeSpanFormFormat,
+} from '../../common/lib/types';
+import { transformToApiFormat } from '../../common/utils/date-time/format';
+import Datepicker from '../../components/datepicker/Datepicker';
+import { PrimaryButton, SecondaryButton } from '../../components/button/Button';
+import { ErrorToast, SuccessToast } from '../../components/notification/Toast';
+import OpeningPeriodDescription from '../description/OpeningPeriodDescription';
+import TimeSpan from '../time-span/TimeSpan';
+import './OpeningPeriodForm.scss';
+
+type OpeningPeriodDataBase = {
+  openingPeriodTitle: string;
+  openingPeriodOptionalDescription: string;
+  openingPeriodBeginDate: string | undefined;
+  openingPeriodEndDate: string | undefined;
+};
+
+interface OpeningPeriodFormData extends OpeningPeriodDataBase {
+  timeSpans: Array<TimeSpanFormFormat> | {}[];
```

Comment:
Just for interest and to improve my typescript skills, can you tell me what that `{}[]` means? :)

Reply:
It accepts an empty object array as well. We need that in as a timeSpan default value. I was struggling with this as well 😅